app/main.py

```diff
@@ -1,3 +1,54 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return (f"Distance: {self.km} kilometers.")
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: ("Distance", int, float)) -> "Distance":
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: ("Distance", int, float)) -> "Distance":
+        if not isinstance(other, Distance):
+            self.km += other
+            return self
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: (int, float)) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km * other, 3))
+
+    def __truediv__(self, other: (int, float)) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: (int, float)) -> bool:
+        if self.km < other:
+            return True
+        return False
+
+    def __gt__(self, other: (int, float)) -> bool:
+        if self.km > other:
+            return True
+        return False
+
+    def __eq__(self, other: (int, float)) -> bool:
+        if self.km == other:
+            return True
+        return False
+
+    def __le__(self, other: (int, float)) -> bool:
```

Comment:
The `__le__` method should also handle comparisons with `Distance` objects, not just numeric types. Update the logic to handle both cases.